### PR TITLE
Add coverage strafter

### LINKED ISSUE #340: Add test coverage on STRAFTER
Add test coverage for all this cases: https://www.w3.org/TR/n-triples/#sec-literals and URIs also

### PR DESCRIPTION
**Pull request by patofp**
This PR adds test coverage to STRAFTER on edge cases, specially on typed and localized strings.

It also sets a way to add same tests to other string functions as STRBEFORE, STRSTARTS, STREND and CONTAINS.

Closes #340

### CODE REVIEW

File: modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala [hunk 1]
```diff
@@ -78,15 +78,21 @@ sealed trait StringVal extends StringLike {
   }
 }
 object StringVal {
-  final case class STRING(s: String)                   extends StringVal
-  final case class DT_STRING(s: String, tag: String)   extends StringVal
+  final case class STRING(s: String)                 extends StringVal
+  final case class DT_STRING(s: String, tag: String) extends StringVal
+  object DT_STRING {
+    def unApply(t: DT_STRING): String = s""""${t.s}"^^${t.tag}"""
```
Comment (pepegar) [inline modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala:84]: I'd rename this to `toString` for example, unapply is the method used under the hood for pattern matching.

File: modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala [hunk 2]
```diff
@@ -78,15 +78,21 @@ sealed trait StringVal extends StringLike {
...
+  }
   final case class LANG_STRING(s: String, tag: String) extends StringVal
-  final case class NUM(s: String)                      extends StringVal
-  final case class VARIABLE(s: String)                 extends StringVal
-  final case object GRAPH_VARIABLE                     extends StringVal { val s = "*g" }
-  final case class URIVAL(s: String)                   extends StringVal
-  final case class BLANK(s: String)                    extends StringVal
-  final case class BOOL(s: String)                     extends StringVal
+  object LANG_STRING {
+    def unApply(l: LANG_STRING): String = s""""${l.s}"${l.tag}"""
```
Comment (pepegar) [inline modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala:88]: Same here.  Also, shouldn't we use @ here?
```suggestion
    def unApply(l: LANG_STRING): String = s""""${l.s}"@${l.tag}"""
```

**Review by pepegar — APPROVED** [review]
Added a couple of comments.  Looks good otherwise!

File: modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala (hunk 1 above)
Comment (pepegar) [reply modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala:84]: or maybe `toRdfLiteral`?

Comment (patofp) [reply modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala:84]: Yeah, that name seems more appropriate! 👍

File: modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala (hunk 2 above)
Comment (patofp) [reply modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala:88]: Hmm, I'll take a look on this.
